Document/0x04f-Testing-Network-Communication.md

```diff
@@ -399,7 +399,16 @@ Some Android and iOS versions do not support some of the recommended cipher suit
 
 ### Static Analysis
 
-Identify all API/web service requests in the source code and ensure that no plain HTTP URLs are used. Make sure that sensitive information is sent over secure channels by using [HttpsURLConnection](https://developer.android.com/reference/javax/net/ssl/HttpsURLConnection.html "HttpsURLConnection") or [SSLSocket](https://developer.android.com/reference/javax/net/ssl/SSLSocket.html "SSLSocket") (for socket-level communication using TLS).
+Since Android 9 (API level 28), cleartext HTTP traffic is blocked by default. However, there are multiple ways in which an application can still send cleartext HTTP traffic:
+
+- Setting the [`android:usesCleartextTraffic`](https://developer.android.com/guide/topics/manifest/application-element#usesCleartextTraffic "Android documentation - usesCleartextTraffic flag") attribute of the `<application>` tag in the AndroidManifest.xml file. Note that this flag is ignored in case a network security config is configured.
+- Configuring the network security config to enable cleartext traffic by setting the cleartextTrafficPermitted attribute to true on `<domain-config>` elements.
+- Using low level libraries (e.g. [`Socket`](https://developer.android.com/reference/java/net/Socket "Socket class") to set up a custom HTTP connection.
+- Using an cross-platform framework, as these typically have their own implementations for HTTP libraries.
+
+Even if one of the above cases is true, it doesn't mean that the application actually sends HTTP traffic. It's possible that an exception is configured, but only HTTPS traffic is used in practice.
+
+Identify all API/web service requests in the source code and ensure that no plain HTTP URLs are used. Make sure that sensitive information is sent over secure channels by using [`HttpsURLConnection`](https://developer.android.com/reference/javax/net/ssl/HttpsURLConnection.html "HttpsURLConnection") or [`SSLSocket`](https://developer.android.com/reference/javax/net/ssl/SSLSocket.html "SSLSocket") (for socket-level communication using TLS).
```

Comment:
Here or in the one of the points above we should highlight that **high-level APIs should be preferred** (HttpsURLConnection over SSLSocket) since there's less the devs can do wrong. This is also recommended by [Apple](https://developer.apple.com/documentation/security/preventing_insecure_network_connections).

> Prefer High-Level Frameworks in Your App
>
>The system enforces ATS when you use the standard URL Loading System. Instances of URLSession automatically negotiate the most secure connection available from the server. 

We must have this recommendation for both platforms in the same test case.

```suggestion
First, you should identify all network requests in the source code and ensure that no plain HTTP URLs are used. Make sure that sensitive information is sent over secure channels by using [`HttpsURLConnection`](https://developer.android.com/reference/javax/net/ssl/HttpsURLConnection.html "HttpsURLConnection") or [`SSLSocket`](https://developer.android.com/reference/javax/net/ssl/SSLSocket.html "SSLSocket") (for socket-level communication using TLS).
```

Reply:
I put "First, " because I think this might be better the first paragraph. A low hanging fruit that can be tested quickly and it's trivial to automate and report.